client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -249,6 +250,78 @@ public void testExists() {
         getAndExistsTest(Request::exists, HttpHead.METHOD_NAME);
     }
 
+    public void testIndicesExist() {
+        String[] indices = randomIndicesNames(1, 10);
+        boolean flatSettings = randomBoolean();
+        boolean includeDefaults = randomBoolean();
+        boolean local = randomBoolean();
+        boolean humanReadable = randomBoolean();
+        boolean ignoreUnavailable = randomBoolean();
+        boolean allowNoIndices = randomBoolean();
+        boolean expandToOpenIndices = randomBoolean();
+        boolean expandToClosedIndices = randomBoolean();
+
+        IndicesOptions indicesOptions = IndicesOptions.fromOptions(
+            ignoreUnavailable,
+            allowNoIndices,
+            expandToOpenIndices,
+            expandToClosedIndices
+        );
+
+        GetIndexRequest getIndexRequest = new GetIndexRequest()
+            .local(local)
+            .humanReadable(humanReadable)
+            .indicesOptions(indicesOptions)
+            .indices(indices)
+            .flatSettings(flatSettings)
+            .includeDefaults(includeDefaults);
+
+        final Request request = Request.indicesExist(getIndexRequest);
+
+        assertEquals("/" + String.join(",", indices), request.getEndpoint());
+        assertEquals(
+            flatSettings,
+            Boolean.valueOf(request.getParameters().get("flat_settings"))
+        );
+        assertEquals(
+            local,
+            Boolean.valueOf(request.getParameters().get("local"))
+        );
+        assertEquals(
+            humanReadable,
+            Boolean.valueOf(request.getParameters().get("human"))
+        );
+        assertEquals(
+            includeDefaults,
+            Boolean.valueOf(request.getParameters().get("include_defaults"))
+        );
+        assertEquals(
+            ignoreUnavailable,
+            Boolean.valueOf(request.getParameters().get("ignore_unavailable"))
+        );
+        assertEquals(
+            allowNoIndices,
+            Boolean.valueOf(request.getParameters().get("allow_no_indices"))
+        );
+        assertEquals(
+            getWildcardValue(expandToOpenIndices, expandToClosedIndices),
+            request.getParameters().get("expand_wildcards")
+        );
```

Comment:
if you use the setRandomX methods like suggested above, they will add the expected params to the provided map, then you can just do one single assertion against the two maps.